### ISSUE TITLE
Cannot insert the value NULL into column 'TransactionDateId', table. (Exercise 2, Task 2)

### ISSUE THREAD

**Issue by slsu0424** [microsoft/MCW-Azure-Synapse-Analytics-and-AI#78]
I am trying to execute this tutorial, to populate the sales table (Exercise 2, Task 2): https://github.com/microsoft/MCW-Azure-Synapse-Analytics-and-AI/blob/master/Hands-on%20lab/HOL%20step-by%20step%20-%20Azure%20Synapse%20Analytics%20and%20AI.md#task-2-populate-the-sale-table

I am getting the following error when trying to execute the pipeline, ASAMCW - Exercise 2 - Copy Sales Data:

{'message':'Job failed due to reason: at Sink 'sale': java.sql.BatchUpdateException: Cannot insert the value NULL into column 'TransactionDateId', table 'Distribution_27.dbo.Table_f7418eb0b0b94ec081fa7919842ff1f9_27'; column does not allow nulls. INSERT fails.\\r\\nThe statement has been terminated.. Details:at Sink 'sale': java.sql.BatchUpdateException: Cannot insert the value NULL into column 'TransactionDateId', table 'Distribution_27.dbo.Table_f7418eb0b0b94ec081fa7919842ff1f9_27'; column does not allow nulls. INSERT fails.\\r\\nThe statement has been terminated.','failureType':'UserError','target':'Data flow1','errorCode':'DFExecutorUserError'}

**Comment by codingbandit** [comment]
I was unable to reproduce this one as well. Ensure that the validation of the environment succeeded in the Before the hand-on lab guide (it seems like you may be missing some data)

![image](https://user-images.githubusercontent.com/3979478/121926973-89322a80-cd0c-11eb-8b77-8f751fe28a0f.png)

**Comment by slsu0424** [comment]
@codingbandit - Interesting - my environment deployed without issue, and I had been able to run through the other tasks in Exercise 2 just fine.

I tried running it with a single parquet file in that larger dataset (, and it produced the same issue.  I ran a query to ensure that there were no nulls in the original parquet file (there are none where TransactionDateId is NULL).

![Screen Shot 2021-06-16 at 5 50 30 PM](https://user-images.githubusercontent.com/25472516/122299186-a8c38180-cecb-11eb-9146-da6d7645c1e7.png)

Is there any other option than having to re-deploy the entire environment?

**Comment by slsu0424** [comment]
@codingbandit - Here is a screenshot querying the parquet file - there are no null values in the TransactionDate field

![Screen Shot 2021-06-16 at 5 57 02 PM](https://user-images.githubusercontent.com/25472516/122299876-8aaa5100-cecc-11eb-86c5-b0f7593cbf3d.png)

**Comment by codingbandit** [comment]
It still seems to be some type of incomplete data. As a test, could you try
a new deployment ensuring you see the successful validation environment
message at the end of the before the hands on lab steps? Also ensure the
ARM deployment has fully completed prior to continuing with the power shell
steps.

**Comment by timahenning** [comment]
Hi @slsu0424, any more thoughts on this issue?

**Comment by slsu0424** [comment]
@timahenning - Thank you for the follow-up.  I was able to get this resolved - turns out there was a small issue with the Data Flow mapping in the target table.  It was by default mapping 'TransactionId' to 'TransactionDateId' in the target table.  I since corrected the source field to 'TransactionDate'.  Hope this helps for anyone else that may encounter this issue.

![Screen Shot 2021-06-22 at 10 30 56 AM](https://user-images.githubusercontent.com/25472516/122943637-2f051b00-d345-11eb-9d9d-a9a2fb958db7.png)
